test/evergreen.yml

```diff
@@ -97,6 +97,19 @@ functions:
           set -o verbose
           echo '#undef HAVE_FTRUNCATE' >> wiredtiger_config.h
     - *make_wiredtiger
+  "compile wiredtiger sanitizer":
+    - command: shell.exec
+      params:
+        working_dir: "wiredtiger/build_posix"
+        shell: bash
+        script: |
+          set -o errexit
+          set -o verbose
+          sh reconf
+          CC=/opt/mongodbtoolchain/v3/bin/clang CXX=/opt/mongodbtoolchain/v3/bin/clang++ PATH=/opt/mongodbtoolchain/v3/bin:$PATH CFLAGS="-ggdb -fPIC -fno-omit-frame-pointer -fsanitize=address" \
+            ../configure ${configure_python_setting|} \
+            --enable-diagnostic --with-builtins=lz4,snappy,zlib
```

Comment:
Is it possible to call `*configure_wiredtiger` and overload `configure_env_vars` and `posix_configure_flags` here? It would be neater, and `*configure_wiredtiger` has taken Windows into consideration, which may benefit future work (e.g. add santizer test coverage for Windows).

Reply:
I was about to ask the same. From what I understand, the initial motivation was to make this function apply to all sanitizer tasks, but from what Luke is saying it seems as if they will require different flags.

Reply:
`compile-asan` task could be a good reference.

Reply:
@tammybailey `--with-builtins=lz4,snappy,zlib` is special for stress testing, compared to non-stress asan tests.

Reply:
@lukech @tammybailey - So I looked into calling `*configure_wiredtiger` with expansion variables and it doesn't work. I looked into documents and by the looks if it we can't call another function from a function as we can call it from a task definition. 

I can copy the logic for windows/linux configuration. I know this won't be super neat either but we are sort of tied by the limitations of Evergreen functions semantics and the fact that `fetch arctifacts` is sometime useless and we still need do compile afterwards (and manually set the function parameters in calling task).

Reply:
Oh sorry I meant at the task level. It looks like e.g. the `configure-combinations` calls the "configure wiredtiger" function to set specific vars and flags, followed by a call to the "make wiredtiger" function.

Reply:
@tammybailey Yes can call it at task context with expansions. But we can't call the task within a task :) . What we can do is make our main task depend on this new special compile task and this is where we hit another problem. Apparently we can't use the output from the special task in our main task because of some issue with embedded static paths in binaries that change from task to task. Have a look at this file for examples where it is done this way and then there is another call for `compile wiredtiger` after fetching the results from compile task! 

No easy option to pick :(

Reply:
I'm not sure what you mean by task within a task? I was thinking something like this:
```
  - name: format-stress-sanitizer-test
    #set a 25 hours timeout
    exec_timeout_secs: 90000
    commands:
      - func: "get project"
      - func: "configure wiredtiger"
        vars:
          configure_env_vars: CC=/opt/mongodbtoolchain/v3/bin/clang CXX=/opt/mongodbtoolchain/v3/bin/clang++ PATH=/opt/mongodbtoolchain/v3/bin:$PATH CFLAGS="-ggdb -fPIC -fno-omit-frame-pointer -fsanitize=address"
          posix_configure_flags: ${configure_python_setting|} --enable-diagnostic --with-builtins=lz4,snappy,zlib
      - func: "make wiredtiger"
      - func: "format test script"
        vars:
          test_env_vars: ASAN_OPTIONS="detect_leaks=1:abort_on_error=1:disable_coredump=0" ASAN_SYMBOLIZER_PATH=/opt/mongodbtoolchain/v3/bin/llvm-symbolizer
          # run for 24 hours ( 24 * 60 = 1440 minutes), stop at first error, use default config
          format_test_script_args: -F -t 60
```

Reply:
Yes that is another option and you can use this as well. But we have multiple tasks that are compiling with these configure options and again the idea is to have a single compile functions for all stress tests that need address sanitization. So it's either between writing a special compile function and call it without any arguments in all relevant task (there are few more such tasks) or calling the generic `compile wiredtiger` with extra arguments in each relevant task.